feature_engine/utils/calculate_conversation_level_features.py

```diff
@@ -0,0 +1,132 @@
+The intention behind this class is to use these modules and define any and all conv level features here. 
+"""
+
+# Importing modules from features
+from features.gini_coefficient import *
+from features.basic_features import *
+from utils.summarize_chat_level_features import *
+
+class ConversationLevelFeaturesCalculator:
+    def __init__(self, chat_data: pd.DataFrame, conv_data: pd.DataFrame) -> None:
+        """
+            This function is used to initialize variables and objects that can be used by all functions of this class.
+
+		PARAMETERS:
+			@param chat_data (pd.DataFrame): This is a pandas dataframe of the chat level features read in from the input dataset.
+            @param conv_data (pd.DataFrame): This is a pandas dataframe of the conversation level features derived from the 
+                                             chat level dataframe.
+        """
+        # Initializing variables
+        self.chat_data = chat_data
+        self.conv_data = conv_data
+
+    def calculate_conversation_level_features(self) -> pd.DataFrame:
+        """
+            This is the main driver function for this class.
+
+		RETURNS:
+			(pd.DataFrame): The conversation level dataset given to this class during initialization along with 
+							new columns for each conv level feature.
+        """
+        # Get gini based features
+        self.get_gini_features()
+        # Get summary statistics by aggregating chat level features
+        self.get_conversation_level_summary_statistics_features()
+        self.get_talkative_member_features()
+        return self.conv_data
+
+    def get_gini_features(self) -> None:
+        """
+            This function is used to calculate the gini index for each conversation 
+            based on the word level and character level information.
+        """
+        # Gini for #Words
+        self.conv_data = pd.merge(
+            left=self.conv_data,
+            right=get_gini(self.chat_data, "num_words"),
+            on=['batch_num', 'round_num'],
+            how="inner"
+        )
+        # Gini for #Characters
+        self.conv_data = pd.merge(
+            left=self.conv_data,
+            right=get_gini(self.chat_data, "num_chars"),
+            on=['batch_num', 'round_num'],
+            how="inner"
+        )
+
+    def get_conversation_level_summary_statistics_features(self) -> None:
+        """
+            This function is used to aggregate the summary statistics from 
+            chat level features to conversation level features.
+            Specifically, it looks at the mean and standard deviations at message and word level.
+        """
+        # Message mean and std
+        self.conv_data = pd.merge(
+            left=self.conv_data,
+            right=get_average(self.chat_data, 'num_messages', 'average_message_count'),
+            on=['batch_num', 'round_num'],
+            how="inner"
+        )
+
+        self.conv_data = pd.merge(
```

Comment:
Here's the bug from earlier -- this calculates mean twice instead of calculating the mean and sd

Reply:
Oh I see. Thanks for identifying and fixing it so soon!